### PR TITLE
Fix dpr<>bart config for RAG

### LINKED ISSUE #8780: Can't load tokenizer for 'facebook/rag-token-base/question_encoder_tokenizer'.
Hi all! I'm getting this error when trying to run the example code:

Can't load tokenizer for 'facebook/rag-token-base/question_encoder_tokenizer'. Make sure that:

- 'facebook/rag-token-base/question_encoder_tokenizer' is a correct model identifier listed on 'https://huggingface.co/models'

- or 'facebook/rag-token-base/question_encoder_tokenizer' is the correct path to a directory containing relevant tokenizer files

### PR DESCRIPTION
**Pull request by patrickvonplaten**
# What does this PR do?

There was a big silent bug between the interaction of DPR and BERT. Bert added a new config parameter that DPR does not have -> so a "normal" dpr config crashes when used with BERT. This is actually a very big bug and was introduces in #8276 as pointed out by @lhoestq - thanks!

Two things went wrong here. 

1) We should be more careful in general when introducing new config parameters and calling them via `config.<new_parameter>` especially for models like BERT that can be used with other configs.

2) The DPR tests should have could that, but instead of using a normal DPR config, a BERT-like DPR config was used in the tests, which is dangerous because it exactly doesn't catch errors like those.

This PR fixes 1) and 2) by calling the config in the case of the newly introduces parameter only with `getattr(config, <param_name>, <default_value>)` **and** adds the config functionality to DPR as well (DPR also should have this functionality over BERT's positional embedding). IMO `getattr(config, <param_name>, <default_value>)` should be used for models like BERT in general because they could be used and wrapped in many different ways.

Also the DPR test is fixed to use a DPR config instead of a BERT config.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors which may be interested in your PR.

### CODE REVIEW

File: tests/test_modeling_dpr.py [hunk 1]
```diff
@@ -104,7 +104,8 @@ def prepare_config_and_inputs(self):
             token_labels = ids_tensor([self.batch_size, self.seq_length], self.num_labels)
             choice_labels = ids_tensor([self.batch_size], self.num_choices)
 
-        config = BertConfig(
```
Comment (patrickvonplaten) [inline tests/test_modeling_dpr.py:107]: This is dangerous went new params are added to Bert, it'll lead to silent errors

File: src/transformers/models/bert/modeling_bert.py [hunk 2]
```diff
@@ -178,7 +178,7 @@ def __init__(self, config):
 
         # position_ids (1, len position emb) is contiguous in memory and exported when serialized
         self.register_buffer("position_ids", torch.arange(config.max_position_embeddings).expand((1, -1)))
-        self.position_embedding_type = config.position_embedding_type
+        self.position_embedding_type = getattr(config, "position_embedding_type", "absolute")
```
Comment (lhoestq) [inline src/transformers/models/bert/modeling_bert.py:181]: maybe add a comment that mentions backward compatibility ?

**Review by lhoestq — APPROVED** [review]
Looks good to me, thanks for fixing it !

File: tests/test_modeling_dpr.py (hunk 1 above)
Comment (LysandreJik) [reply tests/test_modeling_dpr.py:107]: Nice catch

**Review by LysandreJik — APPROVED** [review]
Thank you for fixing this, your fix looks good to me.

**Pull request merged**